thirdparty/pimlico.go

```diff
@@ -209,7 +209,12 @@ func (v *PimlicoVendor) OwnsUpstream(ups *common.UpstreamConfig) bool {
 }
 
 func (v *PimlicoVendor) createClient(ctx context.Context, logger *zerolog.Logger, parsedURL *url.URL) (clients.HttpJsonRpcClient, error) {
-	client, err := clients.NewGenericHttpJsonRpcClient(ctx, logger, "n/a", "n/a", parsedURL, nil)
+	proxyPoolRegistry, err := clients.NewProxyPoolRegistry(common.GetConfig().ProxyPools, logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create proxy pool registry: %v. will use fallback http client", err)
+	}
+
+	client, err := clients.NewGenericHttpJsonRpcClient(ctx, logger, "n/a", "n/a", parsedURL, nil, proxyPoolRegistry)
```

Comment:
We should have a singleton proxyPoolRegistry created on top-level (not on every vendor here), so probably outside of ERPC object (similar to RateLimitersRegistry)

Also we can already fetch and pass the "Pool" only to clients, they don't need to receive the full registry.